internal/handler/web/profile.go

```diff
@@ -20,8 +21,30 @@ func (h *Handler) GetProfile(ctx *gin.Context) {
 		return
 	}
 	ctx.JSON(http.StatusOK, openapi.ProfileResponse{
-		Username: string(user.Username),
-		Roles:    roles,
+		Username:   string(user.Username),
+		Roles:      roles,
+		ChosenName: string(user.ChosenName),
+	})
+}
+
+func (h *Handler) PostProfile(ctx *gin.Context) {
+	user := middleware.GetUser(ctx)
+	update := openapi.ProfileUpdate{}
+	if err := ctx.ShouldBindJSON(&update); err != nil {
+		log.Err(err).Msg("Invalid JSON object")
+		ctx.Status(http.StatusNotAcceptable)
+		return
+	}
+	user.ChosenName = types.ChosenName(update.ChosenName)
+
+	if err := h.profile.SetUserChosenName(&user); err != nil {
```

Comment:
doesn't need to be a pointer
```suggestion
	if err := h.profile.SetUserChosenName(user); err != nil {
```
(then update the function signature)